### DISCUSSION TITLE
No way to add locale files?

### DISCUSSION THREAD

**Discussion by alaanvv** [jesseduffield/lazygit#3657] [Ideas]
Not only would be useful to reach more people but also for editing the texts shown

**Comment by stefanhaller** [comment]
Lazygit is already [translated](https://github.com/jesseduffield/lazygit/tree/master/pkg/i18n) into several other languages, although the translations are pretty out of date, probably. It uses a homegrown translation system for this which makes it hard to keep track of which translations need to be updated.

Work is underway to switch to a more standard translation system, see #3070 and #3649.

Is that what you mean, or were you talking about something else entirely?

**Reply by alaanvv** [reply]
I mean, it would be really great if we could use something like a json to store all text from within the Lazygit, like
```json
{
  "Commits": "~commits~"
}
```
not compiled btw
besides adding an easy way for translators help the project, would make it possible for users to personalize even more, maybe removing completely the title for some windows or adding symbols

but yea, I think you guys probably already thought about that, and if it's not there, probably faced a problem, right?

**Reply by stefanhaller** [reply]
> besides adding an easy way for translators help the project

This is taken care of by #3070.

> would make it possible for users to personalize even more

This is not a goal for me. It's also impossible because we find it important that lazygit ships as a single binary, see https://github.com/jesseduffield/lazygit/issues/3070#issuecomment-1764236314.

**Reply by alaanvv** [reply]
that answers a lot, maybe if the built-in languages still compiled but also had a way to add new on .config it would be cool

I'm closing this, thx for your help, just needed to know if I could edit the text shown (just wanted to make it all lowercase)
